Tests/TuistKitTests/Services/TestServiceTests.swift

```diff
@@ -927,6 +927,124 @@ final class TestServiceTests: TuistUnitTestCase {
         }
     }
 
+    func test_run_tests_caches_passing_targets_when_some_targets_fail() async throws {
+        // Given
+        let projectPath = try temporaryPath().appending(component: "ProjectOne")
+        givenGenerator()
+
+        let scheme = Scheme.test(
+            name: "UnitTests",
+            testAction: .test(
+                targets: [
+                    .test(target: .init(projectPath: projectPath, name: "FrameworkATests")),
+                    .test(target: .init(projectPath: projectPath, name: "FrameworkBTests")),
+                ]
+            )
+        )
+        given(buildGraphInspector)
+            .workspaceSchemes(graphTraverser: .any)
+            .willReturn(
+                [
+                    scheme,
+                ]
+            )
+
+        let graph: Graph = .test(
+            projects: [
+                projectPath: .test(
+                    path: projectPath,
+                    targets: [
+                        .test(
+                            name: "FrameworkATests",
+                            bundleId: "io.tuist.FrameworkATests"
+                        ),
+                        .test(
+                            name: "FrameworkBTests",
+                            bundleId: "io.tuist.FrameworkBTests"
+                        ),
+                    ]
+                ),
+            ]
+        )
+
+        var environment = MapperEnvironment()
+        environment.initialGraph = graph
+        environment.targetTestHashes = [
+            projectPath: [
+                "FrameworkATests": "hash-a",
+                "FrameworkBTests": "hash-b",
+            ],
+        ]
+
+        given(generator)
+            .generateWithGraph(path: .any)
+            .willProduce { path in
+                (path, graph, environment)
+            }
+
+        xcodebuildController.reset()
+
+        // This xcresult bundle has the tests from FrameworkATests failing, and the tests from FrameworkBTests passing
+        let xcResultPath = fixturePath(path: try .init(validating: "Results.xcresult"))
```

Comment:
Let's pass the parser as a dependency and mock the parsed `xcresult` model. This is technically making this unit test into an integration test, which I'd like us to avoid.